src/Features/Core/Portable/MoveStaticMembers/MoveStaticMembersWithDialogCodeAction.cs

```diff
@@ -147,5 +154,106 @@ private static TypeKind GetNewTypeKind(ImmutableArray<ITypeParameterSymbol> type
+                cancellationToken.ThrowIfCancellationRequested();
+
+                // Candidate locations represent a non-perfect match (like incorrect number of params),
+                // which means it likely isn't the member access we're looking for. It's either a reference to
+                // something else, like an overload, in which case we shouldn't change it, or it's an error
+                // by the user, meaning it's ok if we don't change it, as the user will have to fix it anyways
+                if (refLoc.IsCandidateLocation)
+                {
+                    continue;
+                }
+
+                var doc = docEditor.OriginalDocument;
+
+                var syntaxFacts = doc.GetRequiredLanguageService<ISyntaxFactsService>();
+                var refNode = docEditor.OriginalRoot.FindNode(refLoc.Location.SourceSpan, findInsideTrivia: true, getInnermostNodeForTie: true);
+                // track this node in case the doc has changed
+                docEditor.ReplaceNode(refNode, refNode.TrackNodes(refNode));
+
+                // add imports for each reference as there may be multiple locations that we need to add to
+                // if the import is already there, we shouldn't add it again
+                var addImports = doc.GetRequiredLanguageService<IAddImportsService>();
+                docEditor.ReplaceNode(docEditor.OriginalRoot, (node, generator) => addImports.AddImport(
+                    docEditor.SemanticModel.Compilation,
+                    node,
+                    node.GetCurrentNode(refNode)!,
+                    generator.NamespaceImportDeclaration(
+                        newType.ContainingNamespace.ToDisplayString(SymbolDisplayFormats.NameFormat))
+                        .WithAdditionalAnnotations(Simplification.Simplifier.Annotation, Formatting.Formatter.Annotation),
+                    generator,
+                    true,
+                    doc.CanAddImportsInHiddenRegions(),
+                    cancellationToken));
+
+                // now change the actual references to use the new type name
+                // extension methods do not need to be changed as we do not reference the class name
+                if (isExtension)
+                {
+                    continue;
+                }
+
+                // We add a simplifier annotation in both cases (second case is done within the method)
+                // Because sometimes the qualification is wholly or partly unneccessary
+                // static member access should never be pointer or conditional member access
+                if (syntaxFacts.IsNameOfSimpleMemberAccessExpression(refNode))
+                {
+                    var expression = syntaxFacts.GetExpressionOfMemberAccessExpression(refNode.Parent);
+                    if (expression != null)
+                    {
+                        docEditor.ReplaceNode(expression, (node, generator) => generator.TypeExpression(newType)
```

Comment:
i'm not certain how it's safe to replace a general expression with a TypeExpression.  what if hte expression is `(1 + 1)`?

Reply:
All of the member access expressions in this block should strictly be static non-extension members (also not operators or static constructors, as we don't allow movement of those at all), so the expression for the member access should always be a type name. I don't think expressions like `(1 + 1)` could ever call a static non-extension method.